packages/auth/src/providers/cognito/utils/oauth/getRedirectUrl.ts

```diff
@@ -2,22 +2,38 @@
 // SPDX-License-Identifier: Apache-2.0
 import {
 	invalidOriginException,
+	invalidPreferredRedirectUrlException,
 	invalidRedirectException,
 } from '../../../../errors/constants';
 
 /** @internal */
-export function getRedirectUrl(redirects: string[]): string {
-	const redirectUrlFromTheSameOrigin =
-		redirects?.find(isSameOriginAndPathName) ??
-		redirects?.find(isTheSameDomain);
-	const redirectUrlFromDifferentOrigin =
-		redirects?.find(isHttps) ?? redirects?.find(isHttp);
-	if (redirectUrlFromTheSameOrigin) {
-		return redirectUrlFromTheSameOrigin;
-	} else if (redirectUrlFromDifferentOrigin) {
-		throw invalidOriginException;
+export function getRedirectUrl(
+	redirects: string[],
+	preferredSignOutUrl?: string,
+): string {
+	if (preferredSignOutUrl) {
+		const redirectUrl = redirects?.find(
+			redirect => redirect === preferredSignOutUrl,
+		);
+		if (!redirectUrl) {
+			throw invalidPreferredRedirectUrlException;
+		}
+
+		return redirectUrl;
+	} else {
```

Comment:
nit: you meant the following?

```suggestion
	if (preferredSignOutUrl) {
		if (redirects?.includes(preferredSignOutUrl)) {
			return redirectUrl;
		}

		throw invalidPreferredRedirectUrlException;
	} else {
```

Reply:
That reads well, thanks, I'll update it.